src/rest/util.ts

```diff
@@ -37,15 +38,21 @@ export const handleError = (
 };
 
 /**
- * Create a new token for authentication. Requires the
- * `AUTH_SEED` environment variable to be set to obfuscate
- * the values of the tokens generated.
+ * Create a unique new token for authentication. Connects to the remote
+ * database.
  * @returns the token.
  */
-export const createToken = (): string => {
-	return createHash("sha256")
-		.update(`${new Date().toISOString()}${process.env.AUTH_SEED}`)
-		.digest("hex");
+export const createToken = async (): Promise<string> => {
+	const generateHash = () => createHash("sha256").update(`${randomUUID()}`).digest("hex");
+	const { _col, _connection } = await getCollection<User>("user");
+
+	let token = generateHash();
+	while ((await _col.findOne({ token: token })) !== null) {
```

Comment:
This does not need to be a while loop. `findOne` searches all documents in a collection for a match, so the while loop doesn't do anything here. This should be an if statement instead.